src/main/webapp/app/code-editor/service/code-editor-repository.service.ts

```diff
@@ -37,21 +40,43 @@ export interface IBuildLogService {
 
 @Injectable({ providedIn: 'root' })
 export class CodeEditorRepositoryService extends DomainDependentEndpoint implements ICodeEditorRepositoryService {
-    constructor(http: HttpClient, jhiWebsocketService: JhiWebsocketService, domainService: DomainService) {
+    constructor(http: HttpClient, jhiWebsocketService: JhiWebsocketService, domainService: DomainService, private conflictService: CodeEditorConflictStateService) {
         super(http, jhiWebsocketService, domainService);
     }
 
-    isClean = () => {
-        return this.http.get<any>(this.restResourceUrl!).map(data => ({ isClean: data.isClean }));
+    getStatus = () => {
+        return this.http.get<any>(this.restResourceUrl!).pipe(
+            this.handleErrorResponse<{ repositoryStatus: string }>(),
+            tap(({ repositoryStatus }) => {
+                if (repositoryStatus !== CommitState.CONFLICT) {
+                    this.conflictService.notifyConflictState(GitConflictState.OK);
+                }
+            }),
+        );
     };
 
     commit = () => {
-        return this.http.post<void>(`${this.restResourceUrl}/commit`, {});
+        return this.http.post<void>(`${this.restResourceUrl}/commit`, {}).pipe(this.handleErrorResponse());
     };
 
     pull = () => {
-        return this.http.get<void>(`${this.restResourceUrl}/pull`, {});
+        return this.http.get<void>(`${this.restResourceUrl}/pull`, {}).pipe(this.handleErrorResponse());
+    };
+
+    // TODO: Maybe we don't have to check for the error here?
+    resetRepository = () => {
+        return this.http.post<void>(`${this.restResourceUrl}/reset`, {});
     };
+
+    private handleErrorResponse = <T>(): UnaryFunction<Observable<T>, Observable<T>> =>
```

Comment:
This is the most important part of the client implementation:
All responses are checked for the conflict state (409). If a conflict is encountered a service is triggered that components of the code-editor can subscribe to to receive info about conflicts.

I tried a couple of different implementations, but as the conflict will happen very rarely, I tried to move out as much logic from the components into the service (otherwise `ngOnChanges` of the components would be clustered with new checks for this specific case).